lib/rouge/lexers/powershell.rb

```diff
@@ -131,34 +131,43 @@ class Powershell < RegexLexer
         rule %r/[:,]/, Punctuation
       end
 
-      state :hasht do
-        rule %r/\s+/, Text::Whitespace
-        rule %r/\}/, Punctuation, :pop!
+      state :expr do
+        mixin :comments
         rule %r/"/, Str::Double, :dq
         rule %r/'/, Str::Single, :sq
+        rule %r/[{]/, Punctuation, :brace
+      end
+
+      state :hasht do
+        rule %r/\}/, Punctuation, :pop!
         rule %r/\w+/, Name::Other
         rule %r/=/, Operator
         rule %r/,/, Punctuation
+        mixin :expr
         mixin :variable
       end
 
       state :array do
         rule %r/\s+/, Text::Whitespace
         rule %r/\)/, Punctuation, :pop!
-        rule %r/"/, Str::Double, :dq
-        rule %r/'/, Str::Single, :sq
         rule %r/,/, Punctuation
+        mixin :expr
         mixin :variable
       end
 
+      state :brace do
+        rule %r/[}]/, Punctuation, :pop!
+        mixin :root
+      end
+
       state :bracket do
         rule %r/\]/, Punctuation, :pop!
-        rule %r/[A-Za-z]\w+\./, Name::Constant
```

Comment:
I'm a little uncomfortable with a revision of the lexer reducing the level of semantic meaning in the tokens. If a user is concerned about the particular colours being used, isn't the better solution to allow them to adjust their stylesheet?

Reply:
I don't think that's necessarily reasonable - `Name::Constant` is a token that is meant to pop out of the wall of text, and this... is not a constant. It's... a shellword at best.